src/lib/routes/admin-api/project/api-token.ts

```diff
@@ -0,0 +1,214 @@
+                        200: createResponseSchema('apiTokensSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'post',
+            path: PATH,
+            handler: this.createProjectApiToken,
+            permission: CREATE_PROJECT_API_TOKEN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Projects'],
+                    operationId: 'createProjectApiToken',
+                    requestBody: createRequestSchema('createApiTokenSchema'),
+                    responses: {
+                        201: resourceCreatedResponseSchema('apiTokenSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'delete',
+            path: PATH_TOKEN,
+            handler: this.deleteProjectApiToken,
+            acceptAnyContentType: true,
+            permission: DELETE_PROJECT_API_TOKEN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Projects'],
+                    operationId: 'deleteProjectApiToken',
+                    responses: {
+                        200: emptyResponse,
+                    },
+                }),
+            ],
+        });
+    }
+
+    async getProjectApiTokens(
+        req: IAuthRequest,
+        res: Response<ApiTokensSchema>,
+    ): Promise<void> {
+        const { user } = req;
+        const { projectId } = req.params;
+        const projectTokens = await this.accessibleTokens(user, projectId);
+        this.openApiService.respondWithValidation(
+            200,
+            res,
+            apiTokensSchema.$id,
+            { tokens: serializeDates(projectTokens) },
+        );
+    }
+
+    async createProjectApiToken(
+        req: IAuthRequest,
+        res: Response<ApiTokenSchema>,
+    ): Promise<any> {
+        const createToken = await createApiToken.validateAsync(req.body);
+        const { projectId } = req.params;
+        if (!createToken.project) {
+            createToken.project = projectId;
+        }
+
+        if (
+            createToken.projects.length === 1 &&
+            createToken.projects[0] === projectId
+        ) {
+            const token = await this.apiTokenService.createApiToken(
+                createToken,
+                extractUsername(req),
+            );
+            this.openApiService.respondWithValidation(
+                201,
+                res,
+                apiTokenSchema.$id,
+                serializeDates(token),
+                { location: `api-tokens` },
+            );
+        }
+    }
+
+    async deleteProjectApiToken(
+        req: IAuthRequest<ProjectTokenParam>,
+        res: Response,
+    ): Promise<void> {
+        const { user } = req;
+        const { projectId, token } = req.params;
+        const storedToken = (await this.accessibleTokens(user, projectId)).find(
```

Comment:
Service level responsibility? Seems like this controller is performing business logic around the deletion.